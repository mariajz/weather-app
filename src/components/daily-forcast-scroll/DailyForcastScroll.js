import React from 'react';
import { StyledText } from '../../commons/styles';
import { CustomIcon } from '../../commons/visual-elements';
import WeatherCard from '../weather-card';
import {
    CalenderWrapper,
    DailyForcastScrollWrapper,
    HorizontalScroll,
    HorizontalScrollWrapper,
} from './DailyForcastScroll.style';

const DailyForcastScroll = ({ dailyForcastScrollProps }) => {
    const { forecastData } = dailyForcastScrollProps;

    return (
        <DailyForcastScrollWrapper>
            <CalenderWrapper>
                <CustomIcon iconName="calendar" />
                <StyledText marginLeft={5} size={14}>
                    Today's forcast
                </StyledText>
            </CalenderWrapper>
            <HorizontalScrollWrapper>
                <HorizontalScroll
                    horizontal
                    data={forecastData}
                    showsHorizontalScrollIndicator={false}
                    renderItem={({ item }) => {
                        return <WeatherCard weatherItem={item} />;
                    }}
                />
            </HorizontalScrollWrapper>
        </DailyForcastScrollWrapper>
    );
};

export default DailyForcastScroll;
