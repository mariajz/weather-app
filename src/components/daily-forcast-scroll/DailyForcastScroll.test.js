import { render } from '@testing-library/react-native';
import React from 'react';
import DailyForcastScroll from './DailyForcastScroll';

const defaultProps = {
    dailyForcastScrollProps: {
        forcastData: [
            { condition: 'Clear', temp: 24.9, time: '12 AM', weatherImage: 10 },
            { condition: 'Clear', temp: 24.4, time: '1 AM', weatherImage: 10 },
            { condition: 'Clear', temp: 24, time: '2 AM', weatherImage: 10 },
        ],
    },
};
describe('DailyForcastScroll', () => {
    it('should render DailyForcastScroll', () => {
        const container = render(<DailyForcastScroll {...defaultProps} />);

        expect(container).toMatchSnapshot();
    });
});
