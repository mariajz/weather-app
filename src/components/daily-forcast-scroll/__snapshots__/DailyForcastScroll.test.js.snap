// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyForcastScroll should render DailyForcastScroll 1`] = `
<View
  style={
    {
      "height": "35%",
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 10,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../assets/icons/calendar.png",
        }
      }
      style={
        {
          "height": 25,
          "width": 25,
        }
      }
      testID="icon"
      tintColor="white"
    />
    <Text
      marginLeft={5}
      size={14}
      style={
        {
          "color": "white",
          "fontSize": 14,
          "marginLeft": 5,
        }
      }
    >
      Today's forcast
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "width": "100%",
      }
    }
  >
    <RCTScrollView
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      stickyHeaderIndices={[]}
      style={
        {
          "overflow": "visible",
        }
      }
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
